src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -224,10 +225,13 @@ private string GetRootPathWithGuid_NoLock()
             }
 
             // Load the assembly.
-            var assemblyDefinition = AssemblyDefinition.ReadAssembly(assemblyLocation, new ReaderParameters() { AssemblyResolver = new RoslynAssemblyResolver(compilation) });
+            // TODO: Use a different PEFile ctor overload that allows us to reuse the PEReader or Stream already created by Roslyn.
+            var pefile = new PEFile(assemblyLocation, PEStreamOptions.PrefetchEntireImage);
 
             // Initialize a decompiler with default settings.
-            var decompiler = new CSharpDecompiler(assemblyDefinition.MainModule, new DecompilerSettings());
+            // TODO: Use language version currently used by the project.
```

Comment:
📝 This won't be necessary. As #24349 progresses and we favor the original source over decompiled source in more cases, the language version shown for these files will naturally vary from the current project. I don't want to define the Navigate to Decompiled Source in such a way that users expect the language version to match the current project because it will lead to questions/bugs about mismatched versions down the road.

Reply:
That's fine... it was just an idea I had this morning... 😃 I will remove this comment.